### ISSUE TITLE
Can't use extra tables on select while using the CTEBuilder

### ISSUE THREAD

**Issue by slok** [huandu/go-sqlbuilder#160]
Hi!

Thanks for the latest addition of the [CTEBuilder](https://pkg.go.dev/github.com/huandu/go-sqlbuilder#CTEBuilder) :clap:.

I'm having a problem when using it along with more tables, if I want to use the CTE table with another table, when using the CTE table, it wipes the other tables from the select. I don't know if this is a bug or my ignorance of not knowing how to do it, I tried searching for examples but didn't found one.

Let me add an example ([groupwise-max](https://mysql.rjweb.org/doc.php/groupwise_max)) to explain better what I'm having trouble with ([Live example](https://go.dev/play/p/3u0JjDt1y94)):

```golang
package main

import (
	"fmt"

	"github.com/huandu/go-sqlbuilder"
)

type sqlService struct {
	ID      string `db:"s.id"`
	Name    string `db:"s.name"`
	EnvName string `db:"e.name"`
}

var serviceSQLBuilder = sqlbuilder.NewStruct(&sqlService{})

func main() {
	// Create virtual table that holds the services with the latest deployment.
	maxSB := sqlbuilder.NewSelectBuilder()
	maxSB = maxSB.Select(maxSB.As("MAX(d.updated_at)", "max_updated_at"), "d.service_id").From("deployment d")
	maxSB.GroupBy("d.service_id")
	cteb := sqlbuilder.With(sqlbuilder.CTETable("d2").As(maxSB))

	// Retrieve data from the virtual table and join with more data.
	sb := serviceSQLBuilder.SelectFrom("deployment d")
	sb.With(cteb).Where(
		sb.And(
			"d2.max_updated_at = d.updated_at",
			"d2.service_id = s.id",
		),
	)
	sb.Join("service s", "d.service_id = s.id")
	sb.Join("environment e", "s.env_id = e.id")

	fmt.Println(sb.String())
}
```

This gives the result query of (`deployment d` missing):

```sql
WITH d2 AS
  (SELECT MAX(d.updated_at) AS max_updated_at,
          d.service_id
   FROM deployment d
   GROUP BY d.service_id)
SELECT s.id,
       s.name,
       e.name
FROM d2
JOIN service s ON d.service_id = s.id
JOIN environment e ON s.env_id = e.id
WHERE (d2.max_updated_at = d.updated_at
       AND d2.service_id = s.id)
```

However what I would want based on the code should be this (note the `deployment d` table after the `FROM d2`):

```sql
WITH d2 AS
  (SELECT MAX(d.updated_at) AS max_updated_at,
          d.service_id
   FROM deployment d
   GROUP BY d.service_id)
SELECT s.id,
       s.name,
       e.name
FROM d2,
     deployment d
JOIN service s ON d.service_id = s.id
JOIN environment e ON s.env_id = e.id
WHERE (d2.max_updated_at = d.updated_at
       AND d2.service_id = s.id)
```

Many thanks!

**Comment by huandu** [comment]
It's a bug. `SelectBuilder` should not overwrite the `FROM` clause if it's set by `From` before `With`.

**Comment by huandu** [comment]
Released in tag v1.28.1.

**Comment by slok** [comment]
That was fast! Many thanks for the fix :pray:
